flaskbb/management/views.py

```diff
@@ -703,78 +718,66 @@ def delete_category(category_id):
+@management.route("/plugins/<path:name>/enable", methods=["POST"])
 @allows.requires(IsAdmin)
-def enable_plugin(plugin):
-    plugin = get_plugin_from_all(plugin)
+def enable_plugin(name):
+    validate_plugin(name)
+    plugin = PluginRegistry.query.filter_by(name=name).first_or_404()
 
     if plugin.enabled:
         flash(_("Plugin %(plugin)s is already enabled.", plugin=plugin.name),
               "info")
         return redirect(url_for("management.plugins"))
 
-    try:
-        plugin.enable()
-        flash(_("Plugin %(plugin)s enabled. Please restart FlaskBB now.",
-                plugin=plugin.name), "success")
-    except OSError:
-        flash(_("It seems that FlaskBB does not have enough filesystem "
-                "permissions. Try removing the 'DISABLED' file by "
-                "yourself instead."), "danger")
+    plugin.enabled = True
+    plugin.save()
 
+    flash(_("Plugin %(plugin)s enabled. Please restart FlaskBB now.",
+            plugin=plugin.name), "success")
     return redirect(url_for("management.plugins"))
 
 
-@management.route("/plugins/<path:plugin>/disable", methods=["POST"])
+@management.route("/plugins/<path:name>/disable", methods=["POST"])
 @allows.requires(IsAdmin)
-def disable_plugin(plugin):
-    try:
-        plugin = get_plugin(plugin)
-    except KeyError:
-        flash(_("Plugin %(plugin)s not found.", plugin=plugin.name), "danger")
+def disable_plugin(name):
+    validate_plugin(name)
+    plugin = PluginRegistry.query.filter_by(name=name).first_or_404()
+
+    if not plugin.enabled:
+        flash(_("Plugin %(plugin)s is already disabled.", plugin=plugin.name),
+              "info")
         return redirect(url_for("management.plugins"))
 
-    try:
-        plugin.disable()
-        flash(_("Plugin %(plugin)s disabled. Please restart FlaskBB now.",
-                plugin=plugin.name), "success")
-    except OSError:
-        flash(_("It seems that FlaskBB does not have enough filesystem "
-                "permissions. Try creating the 'DISABLED' file by "
-                "yourself instead."), "danger")
+    plugin.enabled = False
+    plugin.save()
+    flash(_("Plugin %(plugin)s disabled. Please restart FlaskBB now.",
+            plugin=plugin.name), "success")
 
     return redirect(url_for("management.plugins"))
 
 
-@management.route("/plugins/<path:plugin>/uninstall", methods=["POST"])
+@management.route("/plugins/<path:name>/uninstall", methods=["POST"])
 @allows.requires(IsAdmin)
-def uninstall_plugin(plugin):
-    plugin = get_plugin_from_all(plugin)
-    if plugin.installed:
-        plugin.uninstall()
-        Setting.invalidate_cache()
-
-        flash(_("Plugin has been uninstalled."), "success")
-    else:
-        flash(_("Cannot uninstall plugin."), "danger")
+def uninstall_plugin(name):
+    validate_plugin(name)
+    plugin = PluginRegistry.query.filter_by(name=name).first_or_404()
 
+    plugin.delete()
```

Comment:
I'm wary of this because it only removes the plugin from the blocked list. The next time the application runs, it'll reappear much to the frustration of the user. We should probably include instructions about uninstalling via `pip` on this page. We should probably also add an `uninstall` hook for plugins to implement so they can do any decoupling needed.